### PR TITLE
Feature/58 manifest default branch

### LINKED ISSUE #58: Use default branch when checking for the manifest file
## Acceptance criteria
- [ ] Default branch is used rather than hardcoding master

### LINKED ISSUE #59: Use default branch when checking for the content of manifest.filePath
## Acceptance criteria
- [ ] Default branch is used rather than hardcoding master

### LINKED ISSUE #62: If the manifest cannot be found in a fork the card should not render
## Solution proposal
* If the .manifest.json file is not returned then the card should  not be rendered.

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
Dependent on #55 

## What issue did you implement or fix?
Fixed #58
Fixed #59
Fixed #62

## Summary
- Validates the manifest before creating fork card

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -100,27 +100,34 @@ async function generateForkCards (forkList) {
...
-      const testResults = document.createElement('p')
-      testResults.textContent = `fake test result number ${i} :shipit:`
-      QS(card, '.testResults').appendChild(testResults)
+    const manifest = await getManifest(fork.full_name, fork.default_branch)
+    const thing = await getCodeSnippet(fork.full_name, fork.default_branch)
+
+    if (typeof manifest.language === 'string' || typeof manifest.filePath === 'string') {
+      for (let i = 0; i < 3; i++) {
+        const testResults = document.createElement('p')
+        testResults.textContent = `fake test result number ${i} :shipit:`
```
Comment (te4-david-jensen) [inline public/js/main.js:109]:
```suggestion
        testResults.textContent = `fake test result number ${i}`
```

File: public/js/main.js [hunk 2]
```diff
@@ -100,27 +100,34 @@ async function generateForkCards (forkList) {
 
   await forkList.forEach(async fork => {
     const card = cardTemplate.cloneNode(true)
-
-    for (let i = 0; i < 3; i++) {
-      const testResults = document.createElement('p')
-      testResults.textContent = `fake test result number ${i} :shipit:`
-      QS(card, '.testResults').appendChild(testResults)
+    const manifest = await getManifest(fork.full_name, fork.default_branch)
+    const thing = await getCodeSnippet(fork.full_name, fork.default_branch)
```
Comment (te4-david-jensen) [inline public/js/main.js:104]:
```suggestion
    const codeSnippet = await getCodeSnippet(fork.full_name, fork.default_branch)
```

File: public/js/main.js [hunk 3]
```diff
@@ -100,27 +100,34 @@ async function generateForkCards (forkList) {
...
+
+    if (typeof manifest.language === 'string' || typeof manifest.filePath === 'string') {
+      for (let i = 0; i < 3; i++) {
+        const testResults = document.createElement('p')
+        testResults.textContent = `fake test result number ${i} :shipit:`
+        QS(card, '.testResults').appendChild(testResults)
+      }
+
+      QS(card, 'h3').textContent = fork.full_name
+      QS(card, 'code').textContent = thing
```
Comment (te4-david-jensen) [inline public/js/main.js:114]:
```suggestion
      QS(card, 'code').textContent = codeSnippet
```

**Review by te4-david-jensen — CHANGES_REQUESTED** [review]
Looks good over all, just a few minor changes and then it's ready!

**Review by te4-david-jensen — APPROVED** [review]
Looks great! 👍  :shipit:

**Review by te4-tintin-wihlborg — APPROVED** [review]
Looks good! 👍

**Pull request merged**